src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitSchedulerClient.kt

```diff
@@ -27,115 +26,114 @@ import java.util.Optional
 class VisitSchedulerClient(
 
   @Qualifier("visitSchedulerWebClient") private val webClient: WebClient,
-  @Value("\${visit-scheduler.api.timeout:10s}") val apiTimeout: Duration,
 ) {
-  fun getVisitByReference(reference: String): VisitDto? {
+  fun getVisitByReference(reference: String): Mono<VisitDto> {
     return webClient.get()
       .uri("/visits/$reference")
       .accept(MediaType.APPLICATION_JSON)
       .retrieve()
-      .bodyToMono<VisitDto>().block(apiTimeout)
+      .bodyToMono()
```

Comment:
Not sure about these changes sorry, VisitSchedulerClient know about the client the receiver does not... use the POC this method should return a DTO not a frame work that is the concern of the client.

```
fun getVisitByReference(reference: String): CompletableFuture<VisitDto> {
    return webClient.get()
      .uri("/visits/$reference")
      .accept(MediaType.APPLICATION_JSON)
      .retrieve()
      .bodyToMono<VisitDto>().toFuture()
  }
```

Reply:
As suggested I have updated the code to always return a DTO object except in one of the visitScheduler calls made by PrisonerProfile endpoint where it returns a Mono.